src/setup-binary.ts

```diff
@@ -0,0 +1,87 @@
+import * as core from "@actions/core";
+import * as hc from "@hashicorp/js-releases";
+import * as io from "@actions/io";
+import * as tc from "@actions/tool-cache";
+import * as sys from "./system";
+import cp from "child_process";
+import path from "path";
+import {ok} from "assert";
+
+const BINARY_NAME = "nomad-pack";
+const USER_AGENT = "setup-${BINARY_NAME} (GitHub Actions)";
+
+export async function setupBinary() {
+  const versionSpec = core.getInput("version");
+
+  let binaryPath = await fetchBinary(versionSpec);
+
+  core.info(`Adding ` + BINARY_NAME + ` to PATH.`);
+  core.addPath(binaryPath);
+
+  let binary = await io.which(BINARY_NAME);
+  let binaryVersion = (cp.execSync(`${binary} version`) || "").toString();
+  core.info(binaryVersion);
+
+  core.setOutput("version", parseVersion(binaryVersion));
+}
+
+export async function fetchBinary(versionSpec: string): Promise<string> {
+  const osPlatform = sys.getPlatform();
+  const osArch = sys.getArch();
+  const tmpDir = getTempDir();
+
+  let binaryPath: string;
```

Comment:
non-constant variables are evil, you may want to extract resolution into internal function